### ISSUE TITLE
BatchNormFuncOptions object cant be printed in C++

### ISSUE THREAD

**Issue by izdeby** [pytorch/pytorch#35770]
Repro:
```
auto options = F::BatchNormFuncOptions().weight(w).bias(b).momentum(m).eps(e).training(t);
std::cout << options << std::endl;
```

Expected result: 
Print value for each property.

Actual Result:
```
‘torch::nn::functional::BatchNormFuncOptions’ is not derived from ‘const torch::ExpandingArray<D, T>’
         std::cout << options << std::endl;
                      ^~~~~~~
```
